ssl.go

```diff
@@ -50,6 +50,16 @@ func ssl(o values) (func(net.Conn) (net.Conn, error), error) {
 		return nil, fmterrorf(`unsupported sslmode %q; only "require" (default), "verify-full", "verify-ca", and "disable" supported`, mode)
 	}
 
+	// Set Server Name Indication (SNI), if enabled by connection parameters.
+	// By default SNI is on, any value != "1" disables SNI.
+	if sslsni := o["sslsni"]; sslsni == "" || sslsni == "1" {
+		// Per RFC 6066, do not set SNI if the host is a literal IP address (IPv4
+		// or IPv6).
+		if net.ParseIP(o["host"]) == nil {
```

Comment:
There is no point of doing that check. It won't work for square bracket notation and TLS will do the determination anyway by calling `crypto.tls.hostnameInSNI`

Reply:
Good catch, thank you